Update SPI Engine register map

StancaPop opened the pull request:
## PR Description

Update the SPI Engine register map. Changes include:

- adding the missing registers
- renaming and cosmetic changes.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the code style guidelines
- [x] I have performed a self-review of changes
- [ ] I have compiled all hdl projects and libraries affected by this PR
- [ ] I have tested in hardware affected projects, at least on relevant boards
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe files, Copyright etc)
- [x] I have not introduced new Warnings/Critical Warnings on compilation
- [ ] I have added new hdl testbenches or updated existing ones

sarpadi commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -241,6 +248,21 @@ ENDFIELD
+
 REG
 0x34
 CMD_FIFO_ROOM
```
replace 0x???????? with ((2^CMD_FIFO_ADDRESS_WIDTH) -1)  if possible

for SDO_FIFO_ROOM
replace 0x???????? with ((2^SDO_FIFO_ADDRESS_WIDTH) -1)  if possible

replace 0x00000000 with 0x0
FIELD
[31:0] 0x00000000
SDI_FIFO_LEVEL

StancaPop replied:
Done.

sarpadi commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -294,7 +316,7 @@ CMD_FIFO
 
 FIELD
-[31:0] 0x?????????
+[31:0] 0x????????
```
replace 0x???????? with 0xXXXXXXXX

StancaPop replied:
Done.

sarpadi commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -328,7 +350,7 @@ SDI_FIFO
 
 FIELD
-[31:0] 0x?????????
+[31:0] 0x????????
```
replace 0x???????? with 0xXXXXXXXX

StancaPop replied:
Done.

StancaPop commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -241,6 +248,21 @@ ENDFIELD
+ENDREG
+
+FIELD
+[31:0] 0x00000000
```
Done.

StancaPop commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -311,7 +333,7 @@ SDO_FIFO
 
 FIELD
-[31:0] 0x?????????
+[31:0] 0x????????
```
Done.

StancaPop commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -339,13 +361,28 @@ ENDFIELD
+ENDREG
+
+FIELD
+[31:0] 0x????????
```
Done.

StancaPop commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -339,13 +361,28 @@ ENDFIELD
 
 FIELD
-[31:0] 0x?????????
+[31:0] 0x????????
```
Done.

StancaPop commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -433,3 +470,59 @@ Writing to this register when the SDO FIFO is full has no effect and the written
 discarded. Reading from this register always returns 0x00000000.
 ENDFIELD
 
+############################################################################################
```
Done.

LBFFilho commented on docs/regmap/adi_regmap_spi_engine.txt:
This also happens here. Seems caused by https://github.com/analogdevicesinc/hdl/blob/beee80ad47417260fc2e336390eba51d1aa5a040/docs/regmap/adi_regmap_spi_engine.txt#L3

Returning this to lowercase makes it appear on the generated html, but then we get a build warning. The other regmap txt seem to have an uppercase name too.

When it builds the register map (with lowercase), the warning is
`axi_spi_engine.rst:1: WARNING: Inline literal start-string without end-string.`

When it doesn't build it (uppercase):
`axi_spi_engine.rst:1: WARNING: Inline literal start-string without end-string.`

I suppose there's a list somewhere of regmaps that should be built, and axi_spi_engine is lowercase there.

@gastmaier is looking into this

LBFFilho replied:
(it was not in doctools, the problem was the things pointed out in the his change requests)

gastmaier commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -1,6 +1,6 @@
 TITLE
 SPI Engine (axi_spi_engine)
-axi_spi_engine
+AXI_SPI_ENGINE
```
Need to update also [axi-spi-engine.rst](https://github.com/analogdevicesinc/hdl/blob/9de7990027d698a892ec009bc13a58342a21cc39/docs/library/spi_engine/axi_spi_engine.rst?plain=1#L79) since this field is unique and case sensitive.
Change from
```
.. hdl-regmap::
   :name: axi_spi_engine
```
to
```
.. hdl-regmap::
   :name: AXI_SPI_ENGINE
```

StancaPop replied:
Done this change. Thank you.

gastmaier commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -425,11 +462,67 @@ OFFLOAD0_SDO_FIFO
+ENDREG
+
+FIELD
+[31:0] 'CFG_INFO_2''
```
Missing single quote,
```
[31:0] ''CFG_INFO_2''
```

gastmaier commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -425,11 +462,67 @@ OFFLOAD0_SDO_FIFO
+ENDREG
 
+FIELD
+[31:0] ''CFG_INFO_3'"
```
Double quote instead of single in the end
```
[31:0] ''CFG_INFO_3''
```
those cause the 
```
WARNING: Inline literal start-string without end-string.
```
error, since effectively you opened an inline literal and didn't close it.

gastmaier commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -425,11 +462,67 @@ OFFLOAD0_SDO_FIFO
+ENDREG
+
+FIELD
+[31:0] 'CFG_INFO_1''
```
Missing single quote,
```
[31:0] ''CFG_INFO_1''
```

StancaPop commented:
Fixed the quotation issues, updated the name of axi_spi_engine to uppercase letters in [axi_spi_engine.rst](https://github.com/analogdevicesinc/hdl/blob/update_spi_engine_regmap/docs/library/spi_engine/axi_spi_engine.rst?plain=1#L79)

sarpadi commented:
DATA_WIDTH register description is not correct and not complete. Value should not be 0x8, register is dependent on params

https://github.com/analogdevicesinc/hdl/blob/9ba4c66c63792523bd6ab3b7b475a540aebab122/library/spi_engine/axi_spi_engine/axi_spi_engine.v#L348

sarpadi commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -232,7 +239,7 @@ SYNC_ID
 
 FIELD
-[31:0] 0x00000000
+[31:0] 0x0
```
SYNC_ID is also read from a FIFO like SDO_DATA, SDI_DATA

sarpadi commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -55,7 +55,7 @@ SCRATCH
+[31:0] 0x0
 SCRATCH
 RW
 Scratch register useful for debug.
```
DATA_WIDTH register description is not correct and not complete. Value should not be 0x8, register is dependent on params

https://github.com/analogdevicesinc/hdl/blob/9ba4c66c63792523bd6ab3b7b475a540aebab122/library/spi_engine/axi_spi_engine/axi_spi_engine.v#L348

StancaPop replied:
Done here: https://github.com/analogdevicesinc/hdl/pull/1294/commits/9c5e3b37e5973b952a7d83267d5709effa4d17b8

sarpadi commented on docs/regmap/adi_regmap_spi_engine.txt:
```diff
@@ -166,33 +166,40 @@ IRQ_MASK
 ENDREG
```
Please change ENABLE default value 0x00000001 to 0x1

StancaPop replied:
Done here: https://github.com/analogdevicesinc/hdl/pull/1294/commits/9c5e3b37e5973b952a7d83267d5709effa4d17b8
